Marsiple/Common/Network/MartianApiManager.swift

```diff
@@ -54,16 +54,20 @@ private extension MartianApiManager {
         task.resume()
     }
     
-    static func postData<DataType: Codable>(data: DataType, url: URL) {
+    static func postData<DataType: Codable>(data: DataType, url: URL, success: @escaping (String) -> Void, failure: @escaping (String) -> Void) {
         guard let json = encodeData(data: data) else { return }
```

Comment:
What happens when the result of `encodeData` is `nil`? This will just fall-through, as we don't invoke the `success` or `failure` callbacks.